tests/test_integration_async.py

```diff
@@ -0,0 +1,617 @@
+import unittest
+from unittest.mock import patch
+
+import html5lib
+from django.core import signing
+from django.core.cache import cache
+from django.db import connection
+from django.http import HttpResponse
+from django.template.loader import get_template
+from django.test import AsyncRequestFactory, RequestFactory
+from django.test.utils import override_settings
+
+from debug_toolbar.forms import SignedDataForm
+from debug_toolbar.middleware import DebugToolbarMiddleware, show_toolbar
+from debug_toolbar.panels import Panel
+from debug_toolbar.toolbar import DebugToolbar
+
+from .base import BaseTestCase, IntegrationTestCase
+from .views import regular_view
+
+rf = RequestFactory()
+arf = AsyncRequestFactory()
+
+
+def toolbar_store_id():
+    def get_response(request):
+        return HttpResponse()
+
+    toolbar = DebugToolbar(rf.get("/"), get_response)
+    toolbar.store()
+    return toolbar.store_id
+
+
+class BuggyPanel(Panel):
+    def title(self):
+        return "BuggyPanel"
+
+    @property
+    def content(self):
+        raise Exception
+
+
+@override_settings(DEBUG=True)
+class DebugToolbarTestCase(BaseTestCase):
+    _is_async = True
+
+    async def test_show_toolbar(self):
+        self.assertTrue(show_toolbar(self.request))
```

Comment:
`show_toolbar` doesn't have any async behavior, so I'm not sure why it should be tested?

Reply:
Originally not truly an async behaviour but `self.request` what would be passed into it will become an `ASGIRequest` in async context that check can be good to have in any case ? somewhat future proofing as well? cause some changes in show toolbar could be of `WSGIRequest` and in those cases it can be useful.

Reply:
Ah well, that's true. 

Maybe remove the `async` though since there's no `await`-ing happening, and add a docstring:

```
"""show_toolbar() works with an ASGIRequest too"""
```

Or something like this?